pkg/operator/sync_test.go

```diff
@@ -118,3 +123,191 @@ func TestWaitForDeploymentRollout(t *testing.T) {
 		})
 	}
 }
+
+func TestSyncValidatingWebhooks(t *testing.T) {
+	defaultHook := mapiv1.NewValidatingWebhookConfiguration()
+	referenceHook := mapiv1.NewValidatingWebhookConfiguration()
+	referenceHook.SetName("test")
+
+	testCases := []struct {
+		testCase       string
+		exisingWebhook func() *admissionregistrationv1.ValidatingWebhookConfiguration
+	}{
+		{
+			testCase: "It should create webhook if it does not exsit",
+		},
+		{
+			testCase: "It should update webhookConfiguration if some of their webhooks differ",
+			exisingWebhook: func() *admissionregistrationv1.ValidatingWebhookConfiguration {
+				hook := defaultHook.DeepCopy()
+				hook.Webhooks[0].Name = "test"
+				return hook
+			},
+		},
+		{
+			testCase: "It should update webhooks if their webhook list is missing an element",
+			exisingWebhook: func() *admissionregistrationv1.ValidatingWebhookConfiguration {
+				hook := defaultHook.DeepCopy()
+				hook.Webhooks = hook.Webhooks[:1]
+				return hook
+			},
+		},
+		{
+			testCase: "Edge case: It shoud not update webhooks if some webhooks are removed from the list",
+			exisingWebhook: func() *admissionregistrationv1.ValidatingWebhookConfiguration {
+				hook := defaultHook.DeepCopy()
+				hook.Webhooks = append(hook.Webhooks, hook.Webhooks...)
+				return hook
+			},
+		},
```

Comment:
Would we not want to be able to tell it exactly which webhooks we want?

Reply:
Sorry, the message is out of date, it is actually not a problem now, and the test is covering that.